### PR TITLE
[MAINTENANCE] Uncap `altair` version, and bump minimum version to `4.2.1`. Also uncap `urllib3` version, and bump minimum version to `1.26`

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- Motivation was Pandas 2.0.0 work which saw a `iteritems` error related to `altair`. 
```
>       return object.__getattribute__(self, name)
E       AttributeError: 'Series' object has no attribute 'iteritems' 
```
- Bumps minimum version of `altair==4.2.1` because it includes the changes from this [PR](https://github.com/altair-viz/altair/pull/2683), which is currently blocking compatibility with pandas 2.0.0
- Found out that the `jsonschema` issues that were preventing the version bump are no longer an issue. 
- Was also able to confirm that this fixes DX-450

- For the `urllib3` version bump, the `altair` change was giving this error, which was addressed in `1.26`. 

```
  ------------------------------ Captured stderr call -----------------------------
ERROR:sqlalchemy.pool.impl.SingletonThreadPool:Exception during reset or similar
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/urllib3/connectionpool.py", line 440, in _make_request
    httplib_response = conn.getresponse(buffering=True)
TypeError: getresponse() got an unexpected keyword argument 'buffering'
```

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -871,7 +871,7 @@ def titanic_pandas_data_context_with_v013_datasource_with_checkpoints_v1_with_em
     # Re-enable GE_USAGE_STATS
     monkeypatch.delenv("GE_USAGE_STATS")
 
-    project_path: str = str(tmp_path_factory.mktemp("titanic_data_context"))
+    project_path: str = str(tmp_path_factory.mktemp("titanic_data_context_013"))
```
Comment (Shinnnyshinshin) [inline tests/conftest.py:896]: This temp directory name is repeated. Was causing state-related issues.

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-altair>=4.0.0,<4.2.1 # conflict with jsonschema
+altair>=4.2.1
```
Comment (Shinnnyshinshin) [inline requirements.txt:1]: This Altair 4.2.1 includes [this PR](https://github.com/altair-viz/altair/pull/2683), which is currently blocking compatibility with `pandas 2.0.0`

Comment (anthonyburdi) [reply requirements.txt:1]: Nice find!

File: tests/conftest.py (hunk 1 above)
Comment (anthonyburdi) [reply tests/conftest.py:896]: Awesome find! Thanks for the helpful explanatory comment 🙇

File: tests/test_packaging.py [hunk 3]
```diff
@@ -179,7 +179,7 @@ def test_polish_and_ratchet_pins_and_upper_bounds():
     )
 
     # Polish and ratchet this number down as low as possible
-    assert len(sorted_packages_with_pins_or_upper_bounds) == 76
+    assert len(sorted_packages_with_pins_or_upper_bounds) == 70
```
Comment (anthonyburdi) [inline tests/test_packaging.py:182]: AWESOME!!!

**Review by anthonyburdi — APPROVED** [review]
Great finds, thanks for upgrading these libraries and removing the pins!

**Pull request merged**